template-only-bin/rename-template-app

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env bash
+# -----------------------------------------------------------------------------
+# This script renames the template application in a project.
+# Run this script in a project's root directory.
+#
+# The project name is the name of the folder in your project's root directory. Use
+# lowercase letters and hyphens. Do not use spaces. Underscores may have unexpected side
+# effects. Choose a unique string that will avoid collisions with commonly used words.
+# By default, the application name is `app-rails`.
+#
+# Positional parameters:
+#   current_name (required) – the current name for the application
+#   new_name (required) - the new name for the application
+# -----------------------------------------------------------------------------
+set -euo pipefail
+
+# Helper to get the correct sed -i behavior for both GNU sed and BSD sed (installed by default on macOS)
+# Hat tip: https://stackoverflow.com/a/38595160
+sedi () {
+  sed --version >/dev/null 2>&1 && sed -i -- "$@" || sed -i "" "$@"
+}
+# Export the function so it can be used in the `find -exec` calls later on
+export -f sedi
+
+current_name=$1
+new_name=$2
+default_name="app-rails"
+
+# Debug:
+echo "---------------------------------------------------------------------"
+echo "current_name: ${current_name}"
+echo "new_name: ${new_name}"
+echo
+
+if [[ "${current_name}" == "${new_name}" ]]; then
+  # Debug:
+  echo "No rename required: ${current_name} == ${new_name}"
+  exit 0
+fi
+
+# Note: Keep this list in sync with the files copied in install-template and update-template
+declare -a include_paths
+include_paths=(.github/workflows/ci-app-rails.yml)
+include_paths+=(.grype.yml)
+include_paths+=(app-rails)
+include_paths+=(docker-compose.yml)
+include_paths+=(docker-compose.mock-production.yml)
+include_paths+=(docs/app-rails)
+
+# Loop through the paths to be included in this template.
+for include_path in "${include_paths[@]}"; do
+  # If the application does not use the default name (i.e. it has already been renamed),
+  # change the include path to use the correct current_name.
+  if [[ "${current_name}" != "${default_name}" ]]; then
+    include_path=$(echo "${include_path}" | sed "s/${default_name}/${current_name}/g")
+  fi
+
+  echo "Checking '${include_path}' to rename '${current_name}' to '${new_name}'..."
+
+  # Skip if the path does not exist.
+  if [[ ! -d "${include_path}" ]] && [[ ! -f "${include_path}" ]]; then
+    echo "Skipping ahead. ${include_path} does not exist in this repo"
+    continue
+  fi
```

Comment:
Good call. This is leftover from a previous version when I was calling `rename-template-app` from both `download-and-install-template` and `update-template`. Now that it's only used by `download-and-install-template`, there is no expected use case where `$include_path` should not exist. I will correct in a follow-up PR.

Reply:
I just remembered why I didn't want to do that: Currently, you can use this script outside the context of installing or updating. It just works. If I made it fail and I wanted to call it manually on a project that is missing a file or path for some reason, it would limit its functionality.

Reply:
I see. Maybe add a warning log/message and comment explaining to review the changes and make sure it didn't rename extra things that shouldn't have been renamed? A little worried about people using it on an existing project where there might be lots of code and therefore extra things get renamed. For example, on pfml the app is called pfml, but I expect there are a lot of "pfml" strings in the codebase, or if we had an app called "notifications" or "documents" I suspect things like variables and API endpoints would get renamed accidentally.

Reply:
https://github.com/navapbc/template-application-rails/pull/34